test/index.ts

```diff
@@ -630,10 +630,46 @@ describe('Datastore', () => {
       const key = new entity.Key({
         path: ['Task', 'Test'],
       });
-      assert.strictEqual(
-        datastore.keyToLegacyUrlsafe(key),
-        'agpwcm9qZWN0LWlkcg4LEgRUYXNrIgRUZXN0DA'
-      );
+      const base64EndocdedUrlsafeKey = 'agpwcm9qZWN0LWlkcg4LEgRUYXNrIgRUZXN0DA';
+      // tslint:disable-next-line: no-any
+      (datastore.auth as any).getProjectId = (callback: Function) => {
+        callback(null, 'project-id');
+      };
+      datastore.keyToLegacyUrlsafe(key, (err, urlSafekey) => {
+        assert.strictEqual(urlSafekey, base64EndocdedUrlsafeKey);
+      });
+    });
+
+    it('should convert key to url safe base64 string with location prefix', () => {
+      const key = new entity.Key({
+        path: ['Task', 'Test'],
+      });
+      const locationPrefix = 's~';
+      const base64EndocdedUrlsafeKey =
+        'agxzfnByb2plY3QtaWRyDgsSBFRhc2siBFRlc3QM';
+      // tslint:disable-next-line: no-any
+      (datastore.auth as any).getProjectId = (callback: Function) => {
+        callback(null, 'project-id');
+      };
+      datastore.keyToLegacyUrlsafe(key, locationPrefix, (err, urlSafekey) => {
+        assert.strictEqual(urlSafekey, base64EndocdedUrlsafeKey);
+      });
+    });
+
+    it('should take projectId from auth if not exists', () => {
```

Comment:
Based on the implementation of the function, I do not think this test is needed.

Reply:
removed!!